facebook/__init__.py

```diff
@@ -311,6 +311,26 @@ def get_app_access_token(self, app_id, app_secret, offline=False):
             return self.request("{0}/oauth/access_token".format(self.version),
                                 args=args)["access_token"]
 
+    def get_code_from_token(self, app_id, app_secret, redirect_uri):
+        """
+        Gets a code to be exchanged for a long lived token.
+        Uses a pre-existing server side long lived token to return a code
+        that can be redeemed for a client side long lived token.
+        """
+        args = {'access_token': self.access_token,
+                'client_id': app_id,
+                'client_secret': app_secret,
+                'redirect_uri': redirect_uri}
+
+        try:
+            code = self.request("{0}/oauth/client_code".format(self.version),
+                                args=args)["code"]
+            return code
+        except GraphAPIError:
+            self.access_token = self.extend_access_token(app_id, app_secret)
+            code = self.request("{0}/oauth/client_code".format(self.version),
+                                args=args)["code"]
+
```

Comment:
If a `GraphAPIError` has been raised, why are we calling `extend_access_token` here? Why isn't the `code` object generated here returned?